src/lib.rs

```diff
@@ -441,4 +583,91 @@ mod test {
         assert_eq!(topo.graph.edge_count(), 7);
         assert!(!is_cyclic_directed(&topo.graph));
     }
+
+    #[test]
+    /* One possible expected output
+     *       ┌───────┐
+     *       │   1   │
+     *       └──▲─┬──┘
+     *     ┌────┘ └────┐
+     * ┌───┴───┐   ┌───▼───┐
+     * │   2   ├───►   3   │
+     * └───▲───┘   └───▲───┘
+     *     └────┐ ┌────┘
+     *       ┌──┴─┴──┐
+     *       │   4   │
+     *       └───────┘
+     */
+    fn test_path_graph_with_ciclic() {
+        let topo = diamond_topology();
+        let topo = topo.paths_graph(4);
+
+        let has_edge = |asn1: u32, asn2: u32| topo.has_connection(asn1, asn2);
+
+        println!("{:?}", Dot::new(&topo.graph));
+
+        assert!(!is_cyclic_directed(&topo.graph));
+        assert!(has_edge(4, 2));
+        assert!(has_edge(4, 3));
+
+        if has_edge(2, 3) {
+            assert!(!has_edge(3, 2));
+            assert!(has_edge(2, 1));
+            assert!(has_edge(1, 3));
+        } else if has_edge(3, 2) {
+            assert!(!has_edge(2, 3));
+            assert!(has_edge(3, 1));
+            assert!(has_edge(1, 2));
+        } else {
+            panic!("should have edge between 2 and 3");
+        }
+    }
+
+    #[test]
+    #[ignore]
+    fn test_path_graph_never_generate_ciclic() {
+        let topo = Topology::from_caida(get_caida_data()).unwrap();
+
+        topo.all_asns().into_par_iter().for_each(|asn| {
+            let topo = topo.paths_graph(asn);
+            assert!(!is_cyclic_directed(&topo.graph));
+        });
+    }
+
```

Comment:
It is a very heavy test, I ran this locally for 30 minutes and didn't find any cycle.